File: languages/csharp/Infisical.Sdk/CommandRunner.cs
using System.Text.Json;

namespace Infisical.Sdk;

internal class CommandRunner
{
    private readonly InfisicalHandle _handle;

    internal CommandRunner(InfisicalHandle handle)
    {
        _handle = handle;
    }

    internal T? RunCommand<T>(Command command)
    {
        var req = JsonSerializer.Serialize(command, Converter.Settings);
        var result = InfisicalLibrary.RunCommand(req, _handle);
        return JsonSerializer.Deserialize<T>(result, Converter.Settings);
    }
}

File: languages/go/command_runner.go
package sdk

import (
	"encoding/json"

	"github.com/infisical/go-sdk/internal/cinterface"
)

type CommandRunnerInterface interface {
	RunCommand(command Command) (string, error)
}

type CommandRunner struct {
	client cinterface.ClientPointer
	lib    cinterface.InfisicalLibrary
}

func NewCommandRunner(client cinterface.ClientPointer, lib cinterface.InfisicalLibrary) *CommandRunner {
	return &CommandRunner{
		client: client,
		lib:    lib,
	}
}

func (c *CommandRunner) RunCommand(command Command) (string, error) {
	commandJSON, err := json.Marshal(command)
	if err != nil {
		return "", err
	}

	responseStr, err := c.lib.RunCommand(string(commandJSON), c.client)
	if err != nil {
		return "", err
	}

	return responseStr, nil
}
